course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -99,14 +146,129 @@
 
                 courseInstance['sites'] = ci.sites;
             }
-            courseInstance['members'] = members.count;
 
             return courseInstance;
         };
 
-        dc.courseInstanceId = $routeParams.courseInstanceId;
+        dc.resetForm = function() {
+            dc.formDisplayData = angular.copy(dc.courseInstance);
+        };
 
-        dc.setCourseInstance($routeParams.courseInstanceId);
+        dc.resetFormFromUI = function() {
+            dc.resetForm();
+            dc.resetGlobalAlerts();
+            dc.alerts.global.formReset = {show:true};
+        };
 
+        dc.resetGlobalAlerts = function(scrollToAlerts) {
+            // reposition viewport at top of iframe so we can see
+            // notification messages
+            dc.alerts.global = {};  // reset any existing global alerts
+            // scroll to top of form by default (i.e. if optional scrollToAlerts
+            // argument is not passed in); can disable by setting scrollToAlerts
+            // to false
+            if (scrollToAlerts || scrollToAlerts == null) {
+                scrollTo(0, 0);
+            }
+        };
+
+        dc.submitCourseDetailsForm = function() {
+            dc.courseDetailsUpdateInProgress = true;
+            var postData = {};
+            var url = djangoUrl.reverse(dc.apiProxy,
+                [dc.apiBase + dc.courseInstanceId + '/']);
+            // we could also get these from editable properties on the DOM
+            var fields = [
+                'description',
+                'instructors_display',
+                'location',
+                'meeting_time',
+                'notes',
+                'short_title',
+                'sub_title',
+                'title'
+            ];
+            fields.forEach(function(field) {
+                postData[field] = dc.formDisplayData[field];
```

Comment:
I had the same thought and went with 'change everything' -- given we don't have a confirmation that says "changing the following fields," or show which fields are dirty and have some form instructions letting the user know it won't update everything, I think that user A would expect that if user B changed fields first for that course instance in another browser B, and then user A submits the values in browser A, the course instance should have all of browser A's details. I think this is a big enough issue to make it a separate story/decision, though, as we'd need to communicate what the form was really doing in a way that made sense.